### ISSUE TITLE
Implement lookaround

### ISSUE THREAD

**Issue by mdinger** [mdinger/rules#1]
- lookahead - `foo <?after bar>` matches `foo` in `foobar`
- negative lookahead - `foo <!after bar>` matches `foo` in `foobaz`
- lookbehind - `<?before foo> bar` matches `bar` in `foobar`
- negative lookbehind - `<!before foo> bar` matches `bar` in `sushibar`

An example with both: `<?before foo> bar <?after baz>` matches `bar` in `foobarbaz`

This is different from Perl 6 as noted in [perl_changes](https://github.com/mdinger/rules/blob/master/perl_changes.md) and is subject to change. I'll need to get experience using it and then compare. Perl 6 may have gotten it correct but I'm not sure or convinced. Granted, I don't think deviating from Perl 6 shouldn't be taken lightly.

I have a PR for this but it doesn't work correctly and I have to do a major revision. Not sure when I'll get to it.

**Comment by olorin37** [comment]
It would be great to propagate raku / perl6 regexes to other languages and Rust seems good for the beginning...

**Comment by mdinger** [comment]
I have not worked on this for a very long time and currently have no plans to currently. If someone decides to start working on this, I can add them as a collaborator or something similar so they can start doing cool things with it.

**Comment by olorin37** [comment]
Yep it would be nice to have it, but I also could not find time for such project currently...

Pozdrawiam,
Jakub A. Gramsz
________________________________
